### PR TITLE
Destination Tag Support

### PR DESCRIPTION
**Pull request by GeekArthur**
Signed-off-by: Jingfu Wang <jingfu.wang@coinbase.com>

### Motivation

This can enable rosetta validation (i.e. rosetta cli) support validation of rosetta implementation with destination tag technology 

### Solution

We need to broadcast transaction with destination tag then validate the destination tag from the on-chain transaction

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -447,6 +447,38 @@ func CurrencyBalance(
 	return liveAmount, liveBlock, nil
 }
 
+// CurrencyBalanceBatch returns the balance batch of an account
+// for all currencies at a particular height.
+// It is up to the caller to determine if the retrieved
+// block has the expected hash for the requested index.
+func CurrencyBalanceBatch(
```
Comment (xiaying-peng) [inline utils/utils.go:454]: Why this change is related to dest tag?

Comment (GeekArthur) [reply utils/utils.go:454]: This is related the support of getting all balances of the pre-funded account when currency is nil, the reason we add this is because we use STX to test destination tag and STX rosetta implementation `/account/balance` only support nil currency, which is compliant our rosetta spec. However, out rosetta cli/sdk has the limitation of doesn't support nil currency, add this for both testing STX and scaling rosetta cli/sdk

Comment (xiaying-peng) [reply utils/utils.go:454]: Maybe rename this method to `AllCurrencyBalance`?

Comment (GeekArthur) [reply utils/utils.go:454]: Sure, will update

File: utils/utils.go [hunk 2]
```diff
@@ -474,26 +506,46 @@ func GetAccountBalances(
...
-
-		if err != nil {
-			return nil, err
-		}
-
-		accountBalance := &AccountBalance{
-			Account: balanceRequest.Account,
-			Amount:  amount,
-			Block:   block,
+		if balanceRequest.Currency != nil {
```
Comment (shrimalmadhur) [inline utils/utils.go:509]: Can you add unit tests for this change?

Comment (GeekArthur) [reply utils/utils.go:509]: Yup, will add them
